src/auxiliary/axpby.jl

```diff
@@ -4,10 +4,8 @@
 # special singleton types (immutables) to remove any overhead of multiplication
 # by one or addition by zero.
 
-struct Zero <: Integer
-end
-struct One <: Integer
-end
+const Zero = Val{0}
+const One = Val{1}
 
 const _zero = Zero()
 const _one = One()
```

Comment:
I guess you also need to remove these definitions, and then find where I use `_zero` and `_one` and replace them appropriately.

Reply:
Good point. :) In the revised version `Val{}` is used in a more "canonical" way and in a few places that use `_one`/`_zero` manually-coded dispatch is replaced by "value type" dispatch.
I've kept the constants because in a few places they are used directly.